app/Http/Controllers/SharingController.php

```diff
@@ -41,10 +41,17 @@ public function listSharing()
 				->join('albums', 'album_id', 'albums.id')
 				->orderBy('title', 'ASC')
 				->orderBy('username', 'ASC')
-				->get();
+				->get()
+				->each(function (&$s) {
+					$album = Album::find($s->album_id);
```

Comment:
This Album::find() is triggering me as we already have the most of data required to apply the getFullPath().
We are just missing `parent_id`.
If we also do a select on `parent_id`, I think it should be possible to apply directly getFullPath() on the resulting object.
This would remove all the calls to `Album::find($s->album_id)`.

Idem for the query in line 68.

Reply:
It doesn't seem like we're getting an `Album` object back from that `DB::table` call, it's a collection of arrays with properties. What we can do is include the `title` and `parent_id` and start building the paths from there, but currently we'd still need an album object to call `getFullPath`. Unless make that method a static and allow it to accept a stdObj with `title` and `parent_id`? Thoughts?

Reply:
> Unless make that method a static and allow it to accept a stdObj with `title` and `parent_id`? Thoughts?

That would be my go to. :|

Reply:
@ildyria I pushed a change that makes `getFullPath` a public static and removes the `Album::find` from the sharing controller.